### PR TITLE
Update auto-generated parameterised gate definition in circuit_to_qasm_io

### PR DESCRIPTION
**Pull request by sjdilkes**

### CODE REVIEW

File: pytket/pytket/qasm/qasm.py [hunk 1]
```diff
@@ -843,11 +843,23 @@ def gdef(self, tree: List) -> None:
...
                 ],
             )
-            if circuit_to_qasm_str(comparison_circ) == circuit_to_qasm_str(gate_circ):
-                existing_op = True
+            # checks that each command has same string
+            existing_op = (
+                len(
+                    set(
+                        [
+                            (str(g) == str(c)) == True
```
Comment (cqc-alec) [inline pytket/pytket/qasm/qasm.py:854]:
```suggestion
                            str(g) == str(c)
```

File: pytket/pytket/qasm/qasm.py [hunk 2]
```diff
@@ -843,11 +843,23 @@ def gdef(self, tree: List) -> None:
...
+                    Symbol("param" + str(index) + "/pi") for index in range(len(symbols))  # type: ignore
                 ],
             )
-            if circuit_to_qasm_str(comparison_circ) == circuit_to_qasm_str(gate_circ):
-                existing_op = True
+            # checks that each command has same string
+            existing_op = (
+                len(
+                    set(
+                        [
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply pytket/pytket/qasm/qasm.py:853]: I did the logic wrong basically, thinking == True satisfied it. I'll just update to check the set is len(1) and that it's single value is 1.

[the remark this answers is not in the document]
Comment (sjdilkes) [reply pytket/pytket/qasm/qasm.py]: I've updated to check that the formed set  == {True}

File: pytket/pytket/qasm/qasm.py (hunk 1 above)
Comment (sjdilkes) [reply pytket/pytket/qasm/qasm.py]: done

File: pytket/pytket/qasm/qasm.py [hunk 3]
```diff
@@ -843,12 +843,18 @@ def gdef(self, tree: List) -> None:
...
-
+            # checks that each command has same string
+            existing_op = set(
+                [
+                    str(g) == str(c)
+                    for g, c in zip(
+                        gate_circ.get_commands(), comparison_circ.get_commands()
+                    )
+                ]
+            ) == {True}
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply pytket/pytket/qasm/qasm.py:857]: I knew there was a more pythonic way to do it, yeah that looks good, thanks!

**Pull request merged**